sys/vm/vm_mmap.c

```diff
@@ -1272,19 +1272,44 @@ struct mincore_args {
 int
 sys_mincore(struct thread *td, struct mincore_args *uap)
 {
+	uintcap_t addr = (uintcap_t)uap->addr;
 
 #if __has_feature(capabilities)
+	uintcap_t range_bottom_page = trunc_page(addr);
+	uintcap_t range_top_page = round_page(addr + uap->len);
```

Comment:
Need to be careful about overflow?

Reply:
I feel like these should all be `ptraddr_t` (or in kernel spelling, `vm_offset_t`) rather than uintcap_t as they are just addresses.  In particular, cheri_getbase returns a `ptraddr_t`.

Reply:
IIRC I concluded that `kern_mincore` handles the overflow check, but we could certainly do it redundantly for clarity.

Reply:
If `kern_mincore` handles overflow I think it's fine to not be redundant here.